include/pstl_offload/internal/usm_memory_replacement.h

```diff
@@ -189,13 +200,24 @@ __internal_aligned_alloc(std::size_t __size, std::size_t __alignment)
         if (__sycl_device_shared_ptr __dev = __offload_policy_holder_type::__get_device_ptr(__offload_policy_holder))
         {
             void* __res = __allocate_shared_for_device(std::move(__dev), __size, __alignment);
-            assert((std::uintptr_t(__res) & (__alignment - 1)) == 0);
+            if (__res && __alignment)
```

Comment:
For everything but `bool` I would recommend to not rely on implicit conversions to `bool`. Readability-wise it gives you an idea what kind of variable you are checking right in the if-statement. The only place where conversion to bool worth it (in my opinion) is a generic code
```suggestion
            if (__res != nullptr && __alignment != 0)
```
You can swap if you like. I mean `nullptr != __res && 0 != alignment)`

Reply:
Done.

Can it be part of code guideline, then?